src/v2/Artsy/Analytics/trackingMiddleware.ts

```diff
@@ -39,7 +47,24 @@ export function trackingMiddleware(options: TrackingMiddlewareOptions = {}) {
           // TODO: Pass referrer over to Artwork page if A/B test passes
           // window.sd.routerReferrer = referrer
 
-          const foundExcludedPath = excludePaths.some((excludedPath) => {
+          /**
+           * Store a global reference to the referrer. Since we're in an SPA
+           * context we'll need to use this to track referrers statefully across
+           * pages, since we don't do a hard reload.
+           *
+           * Attaching to the analytics object in the absence of a more
+           * "global" location.
+           *
+           * For our new reaction apps, all tracking goes through this location:
+           * https://github.com/damassi/force/blob/399919f7ef053701f0ed3b20b32dddf0490459b0/src/desktop/assets/analytics.coffee#L41
+           * which will read __artsyClientSideRoutingReferrer and update referrer appropriately.
+           */
+
+          if (clientSideRoutingReferrer) {
+            analytics.__artsyClientSideRoutingReferrer = getFullReferrerUrl()
+          }
```

Comment:
This is the main change in this file. Previously we would only store this for paths that were not excluded from automatic page view tracking (currently only `/artist/id` is excluded because we track that differently), and now we store it on all transitions, which the artist page can [tap into here](https://github.com/artsy/force/pull/5757/files#diff-9b3ae6b1615915781f4569b076adda28R91).

Reply:
Got it, cool! I think it's artwork page (not artist) that needs to do the custom tracking (which is something to do about special re-targeting events for ecommerce only artworks that are linked to pageviews)...or something like that! 😓

Reply:
Ah yeah, artwork! updated comment. I wish we could smooth out that artwork page one-off some how; thats a pretty consistent vector for things to creep in, but hopefully this change makes it a little more better.